src/ui/2d/token.js

```diff
@@ -60,25 +66,96 @@ export class Token extends React.Component {
     onMouseOut: PropTypes.func,
     children: PropTypes.element,
     animationDuration: PropTypes.number,
+    draggable: PropTypes.bool,
+    shouldDrag: PropTypes.func,
+    onDrag: PropTypes.func,
+    onDrop: PropTypes.func,
+    svgRef: PropTypes.object,
   };
 
   static defaultProps = {
     animationDuration: 750,
     template: Square,
   };
 
+  _startDrag = e => {
+    if (this.props.draggable && this.props.shouldDrag(this.getCoords())) {
+      e = e.touches ? e.touches[0] : e;
+      this.setState({
+        ...this.state,
+        dragged: { x: e.pageX, y: e.pageY },
```

Comment:
Where are `x` and `y` used?  It seems like you're effectively using `dragged` as a boolean in this file?

Reply:
Line 96 and 97 :)

Happy new year 🎆 😄

Reply:
Ah missed that. Thanks! :)